contracts/pegged/tokens/SwapCanonicalToken.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+pragma solidity >= 0.8.9;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/**
+ * @title Canonical token support swap with per-bridge token
+ */
+contract SwapCanonicalToken is ERC20, Ownable {
+    mapping(address => uint256) public bridge_cap; // each bridge address -> mint cap
+    // each bridge token contract.balanceOf(this) tracks how much that bridge has already minted
+
+    constructor(
+        string memory name_,
+        string memory symbol_
+    ) ERC20(name_, symbol_) {
+    }
+
+    // update existing bridge token mint cap or add a new bridge token with mint cap
+    // set cap to 0 will disable swapBridgeForCanonical, but swapCanonicalToBridge will still work
+    function setBridgeTokenMintCap(address bridge_token_address, uint256 mint_cap) external onlyOwner {
```

Comment:
nit: solidity naming convention: _bridgeTokenAddress, _mintCap, similar for other places

Reply:
used [trailing underscore](https://docs.soliditylang.org/en/v0.8.10/style-guide.html#avoiding-naming-collisions) for consistency w/ constructor